lib/faraday/error.rb

```diff
@@ -13,7 +13,10 @@ def initialize(exc, response = nil)
         super(exc.message)
         @wrapped_exception = exc
       elsif exc.respond_to?(:each_key)
-        super("the server responded with status #{exc[:status]}")
+        nil_status_message = 'the server responded with a nil status'
```

Comment:
Does it make sense to say the server responded with a nil status if the server did not send a status Header at all?

Reply:
That's a fair point, maybe a more explicit message, something like `status couldn't be derived from server response`?

Reply:
I like it, I'd say move it down into the error itself as well rather than in this base error class